mq/main/bridge/bridge-api/src/main/java/com/sun/messaging/bridge/api/StompFrameMessage.java

```diff
@@ -423,11 +434,52 @@ public ByteBufferWrapper marshall(Object obj) throws IOException {
         }
     }
 
+    static String unescapeSpecialChars(String keyOrValue) {
+        StringBuilder result = new StringBuilder(keyOrValue.length());
+        boolean backslashSeen = false;
+
+        for (int i = 0; i < keyOrValue.length(); i++) {
+            char c = keyOrValue.charAt(i);
+            if (c == '\\') {
+                if (backslashSeen) {
+                    result.append('\\');
+                    backslashSeen = false;
+                } else {
+                    backslashSeen = true;
+                }
+            } else if (backslashSeen) {
+                if (c == 'c') {
+                    result.append(':');
+                } else if (c == 'r') {
+                    result.append('\r');
+                } else if (c == 'n') {
+                    result.append('\n');
+                } else {
+                    // should throw an exception
```

Comment:
Perhaps `new IllegalArgumentException(something involving c)` could be thrown here?
+UT case please.

Reply:
I have added the exception and an unit test case